### PR TITLE
feat(mission_planner): add another initialization function

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

To realize static center line optimizer as follows, mission_planner has to be called statically.
https://github.com/autowarefoundation/autoware.universe/issues/1942

Therefore, I created this PR to ask mission_planner to plan route statically without runtime callback.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp [hunk 1]
```diff
@@ -31,10 +32,13 @@ class PlannerPlugin
 public:
   using RoutePoints = std::vector<geometry_msgs::msg::Pose>;
   using HADMapRoute = autoware_auto_planning_msgs::msg::HADMapRoute;
+  using HADMapBin = autoware_auto_mapping_msgs::msg::HADMapBin;
   using MarkerArray = visualization_msgs::msg::MarkerArray;
 
   virtual ~PlannerPlugin() = default;
   virtual void initialize(rclcpp::Node * node) = 0;
+  virtual void initialize(
+    rclcpp::Node * node, const autoware_auto_mapping_msgs::msg::HADMapBin::ConstSharedPtr msg) = 0;
```
Comment (isamu-takagi) [inline planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp:41]: Namespace can be removed because `using HADMapBin`.

Comment (takayuki5168) [inline planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp:41]:
```suggestion
    rclcpp::Node * node, const HADMapBin::ConstSharedPtr msg) = 0;
```

Comment (takayuki5168) [reply planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp]: https://github.com/autowarefoundation/autoware.universe/pull/2259/commits/4a49ea8245b941cbfa8f6b48ac51dede0c155019
DONE

**Comment by takayuki5168** [comment]
@isamu-takagi If you can, I wanna get your approval today (before you will have days off).

File: planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp (hunk 1 above)
Comment (isamu-takagi) [reply planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp]: It's same for the other two lines.

Comment (takayuki5168) [reply planning/mission_planner/include/mission_planner/mission_planner_plugin.hpp]: https://github.com/autowarefoundation/autoware.universe/pull/2259/commits/764bc2ffc47ff099d4c1dcd0f8b800b2af88cae3
DONE

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Pull request merged**